templates/analysis.yml

```diff
@@ -0,0 +1,123 @@
+  
+  - script: cd $(System.DefaultWorkingDirectory)/${{ parameters.workDir }} && pwd && ls
+    displayName: Show path and contents
+
+  - script: cd $(System.DefaultWorkingDirectory)/${{ parameters.workDir }} && npm ci
+    displayName: Install dependencies for ${{ parameters.microservice }} --> Task
+
+  - script: cd $(System.DefaultWorkingDirectory)/${{ parameters.workDir }} && npm run build
+    displayName: Build ${{ parameters.microservice }} --> Task
+
+  - task: SonarCloudAnalyze@1
+    displayName: Analyze ${{ parameters.microservice }} --> Task
+# End of Angular-specific Analysis Steps --> if
+
+- ${{ if eq(parameters.angular, false) }}:  # <-- Maven build is the default
+  - task: SonarCloudPrepare@1 # <-- See external requirements note above
+    inputs:
+      SonarCloud: 'SonarCloud'
+      organization: ${{ parameters.sonarOrg }}
+      scannerMode: 'Other'
+      extraProperties: |
+        sonar.projectKey=${{ parameters.sonarKeyPrefix }}${{ parameters.microservice }}
+        sonar.projectName=${{ parameters.sonarNamePrefix }}${{ parameters.microservice }}
+    displayName: Prepare Analysis for ${{ parameters.microservice }} --> Task
+
+  - task: 'Maven@3'
+    inputs:
+      mavenPomFile: ${{ parameters.workDir }}pom.xml
+      publishJUnitresults: true
+      testResultsFiles: '**/surefire-reports/TEST-*.xml'
+      codeCoverageToolOption: 'JaCoCo'
+      codeCoverageClassFilesDirectories: ${{ parameters.coveragePath }}
+      javaHomeOption: 'JDKVersion'
+      jdkVersionOption: '1.11'
+      mavenVersionOption: 'Default'
+      mavenOptions: '-Xmx3072m'
+      mavenAuthenticateFeed: false
+      effectivePomSkip: false
+      sonarQubeRunAnalysis: true
+      sqMavenPluginVersionChoice: 'latest'
+    displayName: Build/Analyze ${{ parameters.microservice }} --> Task
+# End of Maven-specific Analysis steps --> if
+
+- task: SonarCloudPublish@1
+  condition: always()
+  inputs:
+    pollingTimeoutSec: '300'
+  displayName: Publish Analysis of ${{ parameters.microservice }} --> Task
+
+- script: |
+    echo $PARAMETERS_DISCORDCHANNEL
+    echo $PARAMETERS_DISCORDKEY
+  displayName: show discord info
+  
+- task: 'ado-discord-webhook@1' # <-- See external requirements note above
+  condition: always()
+  inputs:
+    channelId: ${{ parameters.discordChannel }}
+    webhookKey: ${{ parameters.discordKey }}
+    name: $(System.TeamProject)
+    messageType: 'content'
+    content: | 
+          **__Analysis Report__**
+          **Status:** $(Agent.JobStatus)
+          **Branch:** $(Build.SourceBranch)
+          **Trigger:** $(Build.Reason)
+          **Message:** $(Build.SourceVersionMessage)
+          **Report:** <${{ parameters.sonarUrlPrefix }}$(Build.SourceBranch)&id=${{ parameters.sonarKeyPrefix }}${{ parameters.microservice }}>
+          **Logs:** <$(System.CollectionUri)$(System.TeamProject)/_build/results?buildId=$(Build.BuildId)&view=logs>
+          **Repo:** <https://github.com/$(Build.Repository.Name)>
```

Comment:
are you sure this is right? doesn't it leave out the org name? rather, i'd expect `Build.Repository.Uri` to work for the whole thing.
also, consider constructing a link directly to the branch in question, and consider distinguishing between PR builds and CI builds appropriately. maybe for another PR.

Reply:
Updated to Build.Repository.Uri per your suggestion
Added separate discord notification for PR builds, includes separate icons for PR and non-PR builds.